.github/workflows/ci-cd.yml

```
@@ -0,0 +1,31 @@
+          project-root: .
+          script-version: 0.5.0
+    outputs:
+      matrix: ${{ steps.initialize.outputs.matrix }}
+  unit-tests:
+    needs: [initialize]
+    strategy:
+      matrix: ${{ fromJson(needs.initialize.outputs.matrix) }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v4
```

yamllint

[warning] wrong indentation: expected 4 but found 6

(indentation)